### DISCUSSION TITLE
Restore deleted posts with retainDeleted set true

### DISCUSSION THREAD

**Discussion by stefanocerri** [payloadcms/payload#1968] [Feature Requests & Ideas]
Hi guys, it would be awsome if there is a trash button to restore deleted posts.
Thank you!

**Comment by DanRibbens** [comment]
I agree Payload should have a complete concept for soft deleting collections. This is going to have an API aspect as well as the admin UI which you're aluding to it seems.

You could use `deleted` checkbox  or `deletedAt` date field in your collection in your project or as a plugin. It would be much better to have this as a built-in feature that is easy to enable and use out of the box.
